src/Orleans/Serialization/BuiltInTypes.cs

```diff
@@ -1275,6 +1278,34 @@ internal static object CopyType(object obj)
         }
 
         #endregion Type
+    
+        #region CancellationTokenWrapper
+        internal static void SerializeCancellationTokenWrapper(object obj, BinaryTokenStreamWriter stream, Type expected)
+        {
+            var ctw = (CancellationTokenWrapper)obj;
+            var cancelled = ctw.CancellationToken.IsCancellationRequested;
+            if (!cancelled)
+            {
+                CancellationTokenManager.RegisterTokenCallbacks(ctw);
```

Comment:
Lets try to maintain layering. Serialization should only serialize, not call upwards into silo logic (we are trying, as best as we can, to maintain acyclic graph of logical dependencies). 
You can do CancellationTokenManager.RegisterTokenCallbacks when the request is sent, from Outside/Inside grain client SendRequest.